homeassistant/components/nam/sensor.py

```diff
@@ -57,12 +48,12 @@ async def async_setup_entry(
             ent_reg.async_update_entity(entity_id, new_unique_id=new_unique_id)
 
     sensors: list[NAMSensor | NAMSensorUptime] = []
-    for sensor in SENSORS:
-        if getattr(coordinator.data, sensor) is not None:
-            if sensor == ATTR_UPTIME:
-                sensors.append(NAMSensorUptime(coordinator, sensor))
+    for description in SENSORS:
+        if getattr(coordinator.data, description.key) is not None:
```

Comment:
`getattr` without default value will raise an `AttributeError` if the attribute isn't found. Not sure if that is an issue here.

Reply:
`nettigo_air_monitor` library returns `None` for all supported sensors if value is not available. For example:

```
NAMSensors(bme280_humidity=64.3, bme280_pressure=1003, bme280_temperature=26.0, bmp280_pressure=None, bmp280_temperature=None, dht22_humidity=None, dht22_temperature=None, heca_humidity=59.9, heca_temperature=27.1, mhz14a_carbon_dioxide=None, sds011_p1=4, sds011_p2=4, sht3x_humidity=None, sht3x_temperature=None, signal=-57, sps30_p0=None, sps30_p1=None, sps30_p2=None, sps30_p4=None, uptime=103438)
```

Reply:
👍🏻 Didn't know if that was the case. Only saw that it could potentially be an issue

Reply:
Thanks for vigilance 😀